projects/schematics/src/add-spartacus/spartacus-features.ts

```diff
@@ -41,169 +41,287 @@ function configureSpartacusModules(
-        CheckoutOccModule,
-        CostCenterOccModule,
-        // Checkout UI
-        CheckoutLoginModule,
-        CheckoutComponentModule,
-        OrderConfirmationModule,
+      [
+        `// Checkout Core,
+        CheckoutModule.forRoot(),`,
+        'CheckoutOccModule,',
+        'CostCenterOccModule,',
+        `// Checkout UI,
+        CheckoutLoginModule,`,
+        'CheckoutComponentModule,',
+        'OrderConfirmationModule,',
+      ].forEach((content) => {
+        addModuleImport(sourceFile, {
+          import: [
+            {
+              moduleSpecifier: SPARTACUS_CORE,
+              namedImports: [
+                'CheckoutModule',
+                'CheckoutOccModule',
+                'CostCenterOccModule',
+              ],
+            },
+            {
+              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
+              namedImports: [
+                'CheckoutComponentModule',
+                'CheckoutLoginModule',
+                'OrderConfirmationModule',
+              ],
+            },
+          ],
+          content,
+        });
+      });
 
-        // Order
-        OrderHistoryModule,
-        OrderDetailsModule,
-        OrderCancellationModule,
-        OrderReturnModule,
-        ReturnRequestListModule,
-        ReturnRequestDetailModule,
-        ReplenishmentOrderHistoryModule,
-        ReplenishmentOrderDetailsModule,
-        ReplenishmentOrderConfirmationModule,
+      [
+        `// Order,
+        OrderHistoryModule,`,
+        'OrderDetailsModule,',
+        'OrderCancellationModule,',
+        'OrderReturnModule,',
+        'ReturnRequestListModule,',
+        'ReturnRequestDetailModule,',
+        'ReplenishmentOrderHistoryModule,',
+        'ReplenishmentOrderDetailsModule,',
+        'ReplenishmentOrderConfirmationModule,',
+      ].forEach((content) => {
+        addModuleImport(sourceFile, {
+          import: [
+            {
+              moduleSpecifier: SPARTACUS_CORE,
+              namedImports: [],
+            },
+            {
+              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
+              namedImports: [
+                'OrderCancellationModule',
+                'OrderDetailsModule',
+                'OrderHistoryModule',
+                'OrderReturnModule',
+                'ReplenishmentOrderConfirmationModule',
+                'ReplenishmentOrderDetailsModule',
+                'ReplenishmentOrderHistoryModule',
+                'ReturnRequestDetailModule',
+                'ReturnRequestListModule',
+              ],
+            },
+          ],
+          content,
+        });
+      });
 
-        // Page Events
-        NavigationEventModule,
-        HomePageEventModule,
-        CartPageEventModule,
-        ProductPageEventModule,
-        `,
+      [
+        `// Page Events,
+        NavigationEventModule,`,
+        'HomePageEventModule,',
+        'CartPageEventModule,',
+        'ProductPageEventModule,',
+      ].forEach((content) => {
```

Comment:
I dont'get why we forEach over all these lines?
Wouldn't it like try to import all the things <number_of_lines> times?

Reply:
It's because we want to check if the individual imports exist.
Otherwise, we would duplicate them only because the order doesn't match, for example.